### ISSUE TITLE
Need help : generateUrl : how to get the slugs from route translations ?

### ISSUE THREAD

**Issue by glecoz** [schmittjoh/JMSI18nRoutingBundle#61]
Hi,

I'm using JMSI18nRoutingBundle with route translations, plus a language switcher of my own, since the [recipe page](http://jmsyst.com/bundles/JMSI18nRoutingBundle/master/cookbook/language_switcher) for that remains empty.

The route translations are working properly with the slugs I've set :

```
sn_homepage:
    pattern:  /members/{username}
    defaults: { _controller: MSWSocialNetworkBundle:Wall:home }
```

and in route translation files something like :

```
MSW_sn_homepage: /membres/{username}
```

for the french transation of the route.

The routes are working properly when I call them in different languages and a username from my browser.

However, I use the following code in my language switcher controller :

```
$url = $this->generateUrl($previous_route_name, array('_locale' => $form['language_switcher']));
return $this->redirect($url);
```

Then I get an error message stating that the required route parameter username isn't passed, of course it works if I manually pass 'username' => 'someone' in the params array of the generateUrl function.

So, in other words, how could I generate the route in my language switcher controller so that the slugs defined in any route translations could be taken in account ?

Many thanks for any help !
